apps/testingaccessibility/src/pages/api/auth/[...nextauth].ts

```diff
@@ -57,14 +58,17 @@ export const nextAuthOptions: NextAuthOptions = {
         const {getPurchasesForUser} = getSdk()
         const purchases = await getPurchasesForUser(token.id as string)
         token.purchases = purchases
+        session.rules = defineRulesForPurchases(purchases)
```

Comment:
this is the **magic** where we create **rules** based on the **purchases** of a specific user id

rules are refreshed on page load and stored on the session so session should always have the current rules

the rules are "cached" on the session so they can be used with `useSession` to use abilities at runtime with `useCurrentAbility`